cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/WidgetRuntimeService.kt

```diff
@@ -45,10 +46,25 @@ class WidgetRuntimeService(
             widgets[id] = WidgetIndex.create(newConfig, vertx).start()
         } else {
             LOGGER.error("Widget Update / Create | " +
-                    "There is widget with no ID in configuration: $widgetConfig")
+                    "There is no widget with given ID in configuration: $widgetConfig")
+        }
+    }
+
+    private fun internalDelete(action: String, widgetConfig: JsonObject): String? {
+        val id = extractId(widgetConfig)
+        if (id != null) {
+            widgets.remove(id)?.stop()
+            LOGGER.info("Widget $action: $widgetConfig")
+        } else {
+            LOGGER.error("Widget $action | " +
+                "There is no widget with given ID in configuration: $widgetConfig")
         }
+        return id
     }
 
+    private fun extractId(widgetConfig: JsonObject) =
```

Comment:
You can make this as an extension function `getId` for `JsonObject` -> `widgetConfig.getId()`

Reply:
Good one, I'm still not fully used to Kotlin